rule.proto

```diff
@@ -0,0 +1,81 @@
+syntax = "proto2";
+package dv360_excel_plugin;
+
+// A message that represents a scheduled rule for manipulating DV360 entities.
+message Rule {
+  // required
+  // The name of the rule, set by the user.
+  optional string name = 1;
+
+  // required
+  // The id that uniquely identified each rule.
+  optional int32 id = 2;
+
+  // required
+  optional Action action = 3;
+
+  // required
+  optional Schedule schedule = 4;
+}
+
+// A message representing the action applied to a DV360 entity.
+message Action {
+  enum Type {
+    UNSPECIFIED_TYPE = 0;
+    CHANGE_LINE_ITEM_STATUS = 1;
+  }
+
+  // required
+  // The type of action.
+  optional Type type = 1;
```

Comment:
What is Type used for, imaging we need to add a new action increase bidding price, we will add a new Type in Type Enum, add a new entry in oneof params, If it is a 1to1 mapping, seems the Type is useless?

Reply:
It's not totally redundant. Using the generated classes, having a Type field makes determining the kind of action more readable.

`myrule.action.type == Action_Type.CHANGE_LINE_ITEM_STATUS`
vs
`myrule.action.whichParams() == Action_Params.changeLineItemStatusParams`

Reply:
ok, make sense from readability perspective.